src/docbuild/cli/cmd_validate/process.py

```diff
@@ -168,6 +168,43 @@ async def validate_rng(
         )
 
 
+def validate_rng_lxml(
+    xmlfile: Path,
+    rng_schema_path: Path = PRODUCT_CONFIG_SCHEMA,
+) -> tuple[bool, str]:
+    """
+    Validate an XML file against a RELAX NG (.rng) schema using lxml.
+
+    This function uses lxml.etree.RelaxNG, which supports only the XML syntax
+    of RELAX NG schemas (i.e., .rng files, not .rnc files).
+
+    :param xmlfile: Path to the XML file to validate.
+    :param rng_schema_path: Path to the RELAX NG (.rng) schema file.
+    :return: Tuple of (is_valid: bool, error_message: str)
+    """
+    try:
+        # Parse the RELAX NG schema from .rng file
+        with open(rng_schema_path, 'rb') as f:
+            relaxng_doc = etree.parse(f)
```

Comment:
The `with` here is normally not needed, you can simplify it with:


```suggestion
        relaxng_doc = etree.parse(rng_schema_path)
```